### PR TITLE
matter: enable Matter shell commands in debug builds

### PR DESCRIPTION
**Pull request by Damian-Nordic**
1. Enable Matter shell commands for all build types except release, for all Matter samples and applications.
2. Merge similar tests in sample.yaml files.
3. Remove no longer needed configurations that are set in Kconfig.defaults or Matter module's Kconfig.

test-sdk-nrf: PR-375

### CODE REVIEW

**Review by kkasperczyk-no — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -360,11 +360,13 @@ Thread samples
 Matter samples
 --------------
 
+* Enabled Matter shell commands for all build types except ``release``.
```
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:363]:
```suggestion
* Enabled Matter shell commands for all build types except ``release`` in all Matter samples.
```
Just a little bit of additional clarity.

Though CLI is only mentioned in the light switch sample. Should this be only in reference to it?

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Review by wiba-nordic — APPROVED** [review]

File: samples/matter/light_switch/sample.yaml [hunk 2]
```diff
@@ -5,7 +5,6 @@ tests:
   # Excluded in quarantine.yaml to limit resources usage in integration builds
   sample.matter.light_switch.debug:
     build_only: true
-    extra_args: CONFIG_CHIP_LIB_SHELL=y
     integration_platforms:
       - nrf52840dk_nrf52840
       - nrf5340dk_nrf5340_cpuapp
```
Comment (jnsgsr) [inline samples/matter/light_switch/sample.yaml:10]: Maybe add nrf7002dk_nrf5340_cpuapp to integration platforms for consistency.

File: samples/matter/template/sample.yaml [hunk 3]
```diff
@@ -5,7 +5,6 @@ tests:
   # Excluded in quarantine.yaml to limit resources usage in integration builds
   sample.matter.template.debug:
     build_only: true
-    extra_args: CONFIG_CHIP_LIB_SHELL=y
     integration_platforms:
```
Comment (jnsgsr) [inline samples/matter/template/sample.yaml:8]: Maybe add nrf7002dk_nrf5340_cpuapp to integration platforms for consistency.

File: samples/matter/light_switch/sample.yaml (hunk 2 above)
Comment (Damian-Nordic) [reply samples/matter/light_switch/sample.yaml:10]: There is a separate `sample.matter.light_switch.debug.nrf7002dk` test for that, right? I thought that was the convention.

**Review by greg-fer — APPROVED** [review]

File: samples/matter/light_switch/sample.yaml (hunk 2 above)
Comment (Damian-Nordic) [reply samples/matter/light_switch/sample.yaml:10]: I will merge this PR now, as it took some time to get a green CI, but I can change it in a separate PR.

**Pull request merged**